### ISSUE TITLE
CORS: PATCH method is not allowed

### ISSUE THREAD

**Issue by fnevgeny** [jacwright/RestServer#109]
I believe in `corsHeaders()`, in

`header('Access-Control-Allow-Methods: GET,POST,PUT,DELETE,OPTIONS');`

the PATCH method is missing.

Thanks for sharing the very useful code!

**Comment by tyzoid** [comment]
Thanks, do you have an example API endpoint that utilizes the `PATCH` method?

**Comment by fnevgeny** [comment]
PATCH is like PUT, but only for updating a part of an entity. In both cases you pass a JSON object of your choice, but while for PUT the object contains all fields, PATCH - only a subset; typically just one (e.g., changing password of a user without altering his/her name/lastname/etc)

**Comment by tyzoid** [comment]
Yeah, I get what `PATCH` is used for, but I'm trying to figure out if it makes sense to have a slightly different way of handling this type of request, or if it makes more sense to just treat them in the same manner as a `POST` internally.

**Comment by fnevgeny** [comment]
On success, `POST` returns 201, `PATCH` (like `PUT`) [should return 204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH). Other than that, I don't think there should be any special treatment.
